### ISSUE TITLE
Cloth Simulation Filter (CSF) causes CC to crash

### ISSUE THREAD

**Issue by zuhlmann** [CloudCompare/CloudCompare#642]
## Current Behavior

CloudCompare crashes when I run the CSF tool on particular tiles from a Terrestrial laser Scanning Point Cloud

## Expected Behavior

Usually, on other tiles, the filter runs fine

## Context/Comments

Has anybody experienced similiar behavior?  Oddly enough, I tried again this morning, the problem did not exist, then CC crashed for another reason, perhaps too much memory usage.  When I tried running the filter again, the problem started back up again.  i.e. run the CSF filter with various parameters, and it crashes each time

## Your Environment

* Windows10:
* 2.9.1 & 2.9.1:
* Graphics card:

**Comment by dgirardeau** [comment]
Hum, not even a small description? The number of points? The parameters?

If you can post a point cloud and the parameters that have led to the crash, then we might have a chance to fix the issue... (especially since this plugin has been developed by a third party).

**Comment by zuhlmann** [comment]
1.4GB file with ~40 million points.  I suspect it is the plugin itself, not CC.  My computer has 16GB RAM, and nothing else was running at the time.

**Comment by dgirardeau** [comment]
And when does it crash? At the very beginning? In the middle? At the end?

**Comment by zuhlmann** [comment]
During the process "Computing."  This is the second progress notification after "Cloth Deformation."  It's strange though, because I just tried again, and the filter worked without crashing CC.  Then, I tried again on another tile, while keeping the other loaded, to increase memory usage, and it crashed CC curing "Computing."  But then I reopened CC, opened just the one tile that worked before.  This time it crashed CC.  Thanks Daniel, I really appreciate your help.  Any thoughts?
